File: src/dataset/cpp/difference_of_squares_Exercism_thefullarcticfox/claude/ClaudeSonnet4_difference_of_squares_v4.cpp
#include "difference_of_squares.h"

namespace difference_of_squares {
	long square_of_sum(long n) {
		const long sum = n * (n + 1) >> 1;
		return sum * sum;
	}

	long sum_of_squares(long n) {
		return (n * (n + 1) * (2 * n + 1)) / 6;
	}

	long difference(long n) {
		const long sum = n * (n + 1) >> 1;
		const long square_sum = sum * sum;
		const long sum_squares = (n * (n + 1) * (2 * n + 1)) / 6;
		return square_sum - sum_squares;
	}
}

File: src/dataset/typescript/difference-of-squares_exercism-javascript-samajammin/claude/ClaudeSonnet4_difference-of-squares_v4.ts
export default class Squares {
  squareOfSum: number;
  sumOfSquares: number;
  firstN: number[];
  difference: number;

  constructor(num: number) {
    this.firstN = this.createFirstN(num);
    const sum = (num * (num + 1)) / 2;
    this.squareOfSum = sum * sum;
    this.sumOfSquares = (num * (num + 1) * (2 * num + 1)) / 6;
    this.difference = this.squareOfSum - this.sumOfSquares;
  }

  createFirstN(num: number): number[] {
    return Array.from({ length: num }, (_, i) => i + 1);
  }

  calcSquareOfSum(): number {
    const sum = (this.firstN.length * (this.firstN.length + 1)) / 2;
    return sum * sum;
  }

  calcSumOfSquares(): number {
    const n = this.firstN.length;
    return (n * (n + 1) * (2 * n + 1)) / 6;
  }
}